src/main/kotlin/io/sc3/peripherals/posters/PosterPrintData.kt

```diff
@@ -0,0 +1,70 @@
+package io.sc3.peripherals.posters
+
+import io.sc3.library.ext.optString
+import io.sc3.library.ext.putOptString
+import io.sc3.peripherals.config.ScPeripheralsConfig.config
+import io.sc3.peripherals.mixin.MapColorAccessor
+import io.sc3.peripherals.posters.PosterItem.Companion.POSTER_KEY
+import net.minecraft.block.MapColor
+import net.minecraft.nbt.NbtCompound
+import net.minecraft.text.Text
+import kotlin.math.roundToInt
+import kotlin.math.sign
+
+const val MAX_LABEL_LENGTH = 48
+const val MAX_TOOLTIP_LENGTH = 256
+
+fun getDefaultPalette() = MapColorAccessor.getColors().map { it?.color ?: MapColor.CLEAR.color }.toMutableList()
+
+data class PosterPrintData(
+  private val initialLabel: String? = null,
+  var tooltip: String? = null,
+  val colors: MutableList<Int> = MutableList(128 * 128) { 0 },
```

Comment:
Since this is fixed to 128x128, and the palette max size is 64, can we change this from an arraylist to a ByteArray? I'm a little worried with how many of these are going to be around in memory.

As I'm reading further through the code I see that in some places it's a ByteArray (and serialized in nbt as a byte array) and in other places it's an IntArray (and serialized in nbt as an int array) - I think it would be good to keep this consistently a byte array everywhere